### PR TITLE
docs: add etcd backup and restore guide

### PR DESCRIPTION
**Pull request by smira**
Describe full procedure from backups to disaster recovery.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+---
+title: "Disaster Recovery"
+description: "Procedure for snapshotting etcd database and recovering from catastrophic control plane failure."
+---
+
+`etcd` database backs Kubernetes control plane state, so if `etcd` service is unavailable
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:6]: The `etcd`... unavailable,

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+Kubernetes control plane goes down, and the cluster is not recoverable until `etcd` is recovered with contents.
+`etcd` consistency model builds around consensus protocol Raft, so for highly-available control plane clusters,
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:8]: (articles) The `etcd` consistency... the consensus...

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+loss of one control plane doesn't impact cluster health.
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:9]: "one control plane _member_"(?)

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+In general, `etcd` stays up as long as quorum nodes are up.
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:10]: ...stays up as long as a sufficient number of nodes to maintain quorum are up.

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+For three control plane node Talos cluster, this means that the cluster tolerates a failure of any single node,
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:11]: For _a_ three

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+but losing more than one node leads to complete cluster failure.
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:12]: This sounds just a touch _too_ dire.  Maybe: "losing more than one node at the same time" or "complete loss of service"?

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+Because of that, it is important to take backups of `etcd` state to have a snapshot to recover cluster from
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:13]: ... take _routine_ (or maybe regular) backups...

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+Create a consistent snapshot of `etcd` database with `talosctl etcd snapshot` command:
+
+```bash
+$ talosctl -n <IP> etcd snapshot db.snapshot
+etcd snapshot saved to "db.snapshot" (2015264 bytes)
+snapshot info: hash c25fd181, revision 4193, total keys 1287, total size 3035136
+```
+
+This database snapshot can be taken on any healthy control plane node, as all `etcd` instances contain
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:28]: Might add a note that the IP above must be the IP address of one of the controlplane node

Comment (Ulexus) [reply website/content/docs/v0.10/Guides/disaster-recovery.md:28]: and that the filename `db.snapshot` is arbitrary

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+### Disaster Database Snapshot
+
+If `etcd` cluster is not healthy command `talosctl etcd snapshot` might fail.
+In that case, as a last resort, copy the database snapshot from the control plane node:
+
+```bash
+talosctl -n <IP> cp /var/lib/etcd/member/snap/db .
+```
+
+This snapshot might not be fully consistent, but it allows for disaster recovery when last consistent
```
Comment (Ulexus) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:41]: Snapshots pulled this way are, in fact, guaranteed to be consistent (unless I am very mistaken), but they do not (apparently) include the check hash.

Comment (smira) [reply website/content/docs/v0.10/Guides/disaster-recovery.md:41]: yep, thank you!

Comment (smira) [reply website/content/docs/v0.10/Guides/disaster-recovery.md:41]: but I guess if you copy it while `etcd` is running, it won't be consistent, as there's no way writes are synchronized with copy.

Comment (smira) [reply website/content/docs/v0.10/Guides/disaster-recovery.md:41]: I rephrased this sentence a bit.

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+$ talosctl -n <IP> etcd snapshot db.snapshot
+etcd snapshot saved to "db.snapshot" (2015264 bytes)
+snapshot info: hash c25fd181, revision 4193, total keys 1287, total size 3035136
+```
+
+> Note: filename `db.snapshot` is arbitrary.
+
+This database snapshot can be taken on any healthy control plane node (with IP address `<IP>` in the example above),
+as all `etcd` instances contain exactly same data.
+Configure `etcd` snapshots to be created on some schedule to allow point-in-time recovery using the latest snapshot.
```
Comment (rsmitty) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:32]:
```suggestion
It is recommended to configure `etcd` snapshots to be created on some schedule to allow point-in-time recovery using the latest snapshot.
```

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+
+### Disaster Database Snapshot
+
+If `etcd` cluster is not healthy command `talosctl etcd snapshot` might fail.
```
Comment (rsmitty) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:36]:
```suggestion
If `etcd` cluster is not healthy, the `talosctl etcd snapshot` command might fail.
```

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+procedure.
+
+### Latest Etcd Snapshot
+
+Get hold of the latest `etcd` database snapshot.
+If a snapshot is not fresh enough, create a database snapshot (see above),  even if the `etcd` cluster is unhealthy.
+
+### Init Node
+
+Make sure that there are no control plane node with machine type `init`:
```
Comment (rsmitty) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:72]:
```suggestion
Make sure that there are no control plane nodes with machine type `init`:
```

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+`init` node can be converted to `controlplane` type with `talosctl edit mc --on-reboot` command followed
+by node reboot with `talosctl reboot` command.
+
+### Preparing Control Plane Nodes
+
+If some control plane nodes experienced hardware failure, replace them with new nodes.
+Use machine configuration backup to re-create the nodes with the same secret material and control plane settings
+to allow workers to join the recovered control plane.
+
+If a control plane node is healthy but `etcd` isn't, wipe the node `EPHEMERAL` partition to remove the `etcd`
```
Comment (rsmitty) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:92]:
```suggestion
If a control plane node is healthy but `etcd` isn't, wipe the node's `EPHEMERAL` partition to remove the `etcd`
```

File: website/content/docs/v0.10/Guides/disaster-recovery.md [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+
+Now `etcd` service should become healthy on the bootstrap node, Kubernetes control plane components
+should start and control plane endpoint should become available.
+Remaining control plane nodes join `etcd` cluster once control plane endpoint is up.
+
+## Single Control Plane Node Cluster
+
+This guide applies to the single control plane clusters as well.
+In fact, it is much more important to take regular snapshots of the `etcd` database in single control plane node
+case, as loss of the control plane node might make whole cluster irrecoverable without a backup.
```
Comment (rsmitty) [inline website/content/docs/v0.10/Guides/disaster-recovery.md:147]:
```suggestion
case, as loss of the control plane node might render the whole cluster irrecoverable without a backup.
```

**Review by Ulexus — APPROVED** [review]

**Pull request merged**